src/user-guide/writing-workflows/external-triggers.rst

```diff
@@ -207,137 +451,127 @@ To see this, take a look at the job script for one of the downstream tasks:
 
-- ``foo`` spawns out to the runahead limit.
-- ``FAM`` spawns only when ``@upstream`` is satisfied.
-- All associated xtriggers are checked and, as expected, their satisfaction
-  is a prerequisite to task readiness.
+.. literalinclude:: ../../workflows/xtrigger/xrandom/flow.cylc
+   :language: cylc
 
 
 .. _Custom Trigger Functions:
 
-Custom Trigger Functions
-------------------------
+Custom Xtrigger Functions
+-------------------------
 
-Trigger functions are just normal Python functions, with a few special
-properties:
+Xtrigger functions are Python functions with some special requirements.
 
-- They must:
 
-  - Be defined in a module with the same name as the function, unless
-    provided using the ``cylc.xtriggers`` entry point;
-  - be compatible with the same Python version that runs the scheduler
-    (see :ref:`Requirements` for the latest version specification).
-
-- They can be located either:
+Requirements
+^^^^^^^^^^^^
 
-  - In ``<workflow-dir>/lib/python/``;
-  - Anywhere in your ``$CYLC_PYTHONPATH``;
-  - Defined using the ``cylc.xtriggers`` entry point for an installed
-    package - see :ref:`developing.xtrigger.plugins`
+Xtrigger functions must be compatible with the Python version that runs the
+scheduler (see :ref:`Requirements` for the latest version specification).
 
-- They can take arbitrary positional and keyword arguments
-  (except ``sequential``, which is reserved - see :ref:`Sequential Xtriggers`)
-- Workflow and task identity, and cycle point, can be passed to trigger
-  functions by using string templates in function arguments (see below)
-- Integer, float, boolean, and string arguments will be recognized and
-  passed to the function as such
-- If a trigger function depends on files or directories (for example)
-  that might not exist when the function is first called, just return
-- The module can also provide a ``validate`` function for checking configured
-  arguments / keyword arguments, see
-  :ref:`user-guide.xtrigger-validation-functions` for details.
+They must return a Tuple of ``(Boolean, Dictionary)``:
 
-.. note::
+  - ``(False, {})`` - failed: trigger condition not met
+  - ``(True, results)`` - succeeded: trigger condition met
 
-   Trigger functions cannot store data Pythonically between invocations
-   because each call is executed in an independent process in the process
-   pool. If necessary the filesystem can be used for this purpose.
+where ``results`` is a flat (non-nested) dictionary of information to be passed
+to dependent tasks - see :ref:`Xtrigger Results`. Each dictionary key must be
+valid as an
+`environment variable <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html>`_
+name.
 
+They can take arbitrary positional and keyword arguments, except that the
+``sequential`` keyword is reserved for :ref:`sequential xtriggers <Sequential Xtriggers>`.
 
-.. spelling:word-list::
+They cannot store data Pythonically between invocations, because each call
```

Comment:
[Don't really care that much, arguable style point] IMO I think "Custom xtrigger functions" or "custom xtriggers", or even just "xtriggers" would be more desirable than "they".

Reply:
Yes, I was unsure of this myself. I decided a bulleted list of long sentences doesn't look great in this context (it's a more of a book than a powerpoint presentation), but repeating the object on every line seemed too much.

Reply:
OK I've gone with repetition, as the lesser of two evils.